src/Core/PostgreSQL/ConnectionHolder.h

```diff
@@ -40,19 +40,21 @@ class ConnectionHolder
         {
             try
             {
-                connection->getRef().reset();
+                // First try to close the connection gracefully.
+                connection->close();
             }
             catch (...)
             {
+                // If that fails for some reason, hard-reset it.
```

Comment:
Reset calls `~Connection` with also does `Connection::close`, not not clear why should we do like this.

Reply:
I'm a little fuzzy on the expected logic here. The original code calls `pqxx::connection::reset()`, which is in the header but undocumented as far as I can see: https://libpqxx.readthedocs.io/stable/classpqxx_1_1connection.html#details

I've restored the behavior of calling it, but I suspect `close` or something similar might be better. I'm not sure the pros/cons.

Finally, when the connection is reset won't a null ptr be returned to the pool?